app/src/main/java/com/example/shareyourtrip/PostAdapter.java

```diff
@@ -89,15 +137,67 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                     //If favorite is toggled on...
                     if (isChecked) {
                         holder.favButton.setBackgroundResource(R.drawable.ic_favorite);
-                        //todo: Put favorite database stuff here
+
+                        if( (currentContext instanceof SearchActivity) || (currentContext instanceof HomePageActivity) ) {
+                            //todo: Put favorite database stuff here
+                            int pos = holder.getAdapterPosition();
+
+                            // check if item still exists
+                            if (pos != RecyclerView.NO_POSITION) {
+                                Post post = postList.get(pos);
+                                clickedPost = new Post(post);
+
+                                if (!postDAO.insertFavPost(post)) {
+                                    Toast.makeText(currentContext, "This post is already you favortie!", Toast.LENGTH_LONG).show();
+                                }
+                            }
+                        }
                     }
                     //If favorite is toggled off
                     else {
                         holder.favButton.setBackgroundResource(R.drawable.ic_unfavorited);
+
+                        //Unfavorite functionality only works in favorite page
+                        if(currentContext instanceof FavoriteActivity) {
+                            int pos = holder.getAdapterPosition();
+                            if (pos != RecyclerView.NO_POSITION) {
+                                Post post = postList.get(pos);
+                                clickedPost = new Post(post);
+
+                                String[] args = {clickedPost.getId().toString(), FirebaseAuth.getInstance().getCurrentUser().getEmail()};
+                                postDAO.deleteFavPost("postid = '" + clickedPost.getId().toString() + "' and useremail = '" + FirebaseAuth.getInstance().getCurrentUser().getEmail() + "'", null);
+                            }
+                        }
```

Comment:
What if the favorite button is toggled off in the Search or Home activity? Would need "else if( (currentContext instanceof SearchActivity) || (currentContext instanceof HomePageActivity) )"